keyservice/server.go

```diff
@@ -0,0 +1,113 @@
+package keyservice
+
+import (
+	"go.mozilla.org/sops/kms"
+	"go.mozilla.org/sops/pgp"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+type Server struct{}
+
+func (ks *Server) encryptWithPgp(key *GpgKey, plaintext []byte) ([]byte, error) {
+	pgpKey := pgp.NewMasterKeyFromFingerprint(key.Fingerprint)
+	err := pgpKey.Encrypt(plaintext)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(pgpKey.EncryptedKey), nil
+}
+
+func (ks *Server) encryptWithKms(key *KmsKey, plaintext []byte) ([]byte, error) {
+	var ctx map[string]*string
+	for k, v := range key.Context {
+		ctx[k] = &v
+	}
+	kmsKey := kms.MasterKey{
+		Arn:               key.Arn,
+		Role:              key.Role,
+		EncryptionContext: ctx,
+	}
+	err := kmsKey.Encrypt(plaintext)
```

Comment:
this is that object-oriented code smell I'm talking about - we're using the MasterKey (interface) here to encrypt the data key, but we're not returning the encrypted data key via this method - instead we're setting a field of the MasterKey (struct) itself to the encrypted data key and then later using that value.

Reply:
Is there a reason why it's implemented this way, versus returning `EncryptedKey` as the output of `kmsKey.Encrypt(plaintext)` ?

Reply:
Yes. The encrypted data key has to be in the MasterKey, because we have to keep track of what MasterKey encrypted the data key. The encrypted data key has to end up in the output file, so we need to keep that information around, somehow. I don't disagree that this being part of the MasterKey interface is bad, but this kind of was the way it worked before with `ToMap` (which we could remove now). This should probably go in another PR and not in this one.